stringtemplate4/src/main/java/org/jdbi/v3/stringtemplate4/StringTemplateSqlLocator.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.stringtemplate4;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import net.jodah.expiringmap.ExpirationPolicy;
+import net.jodah.expiringmap.ExpiringMap;
+import org.stringtemplate.v4.ST;
+import org.stringtemplate.v4.STGroup;
+import org.stringtemplate.v4.STGroupString;
+
+/**
+ * Locates SQL in <code>.sql.stg</code> StringTemplate group files on the classpath.
+ */
+public class StringTemplateSqlLocator {
+    private static final Map<Class<?>, STGroup> CACHE = ExpiringMap.builder()
+            .expiration(10, TimeUnit.MINUTES)
+            .expirationPolicy(ExpirationPolicy.ACCESSED)
+            .entryLoader(o -> readStringTemplateGroup((Class) o))
+            .build();
+
+    private static final Charset UTF_8 = Charset.forName("UTF-8");
+    private static final String TEMPLATE_GROUP_EXTENSION = ".sql.stg";
+
+    private StringTemplateSqlLocator() {
+    }
+
+    /**
+     * Locates SQL for the given type and name. Example: Given a type <code>com.foo.Bar</code> and a name of
+     * <code>baz</code>, loads a StringTemplate group file from the resource named <code>com/foo/Bar.sql.stg</code> on
+     * the classpath, and returns the template with the given name from the group.
+     *
+     * @param type the type that "owns" the given StringTemplate group file. Dictates the filename of the
+     *             StringTemplate group file on the classpath.
+     * @param name the template name in the StringTemplate group.
+     * @return the located SQL.
+     */
+    public static ST findStringTemplate(Class<?> type, String name) {
```

Comment:
This method differs from the one in ST3 since ST4 doesn't let you get the original template string back out.

I feel like the ST3 one should also return a template instance, instead of a string, since a template could reference another template in the group.